### PR TITLE
D4Science fixes

### PR DESCRIPTION
**Pull request by enolfc**
# Summary

Bring several fixes to make the hub work on GCP:
- Make the use of the sidecar for mounting the workspace, let it be handled by the single-user image instead.
- Upgrade the image to the latest z2j hub images (v3.2.1)
- Fix the authenticator code to fetch the D4Science context at the right place
- Bring the ophidia support into the spawner

---

**Related issue :**

### CODE REVIEW

**Comment by enolfc** [comment]
Found an issue, so moving to draft for the time being

**Comment by enolfc** [comment]
> Found an issue, so moving to draft for the time being

Was a misconfiguration of the helm, not related with the code... so good to be reviewed

File: egi_notebooks_hub/d4science.py [hunk 1]
```diff
@@ -449,9 +483,53 @@ def profile_list(self, spawner):
...
+        if "ophidia" in chosen_profile:
+            ophidia_mounts = [
+                m for m in spawner.volume_mounts if m["name"] != "workspace"
+            ]
+            ophidia = {
+                "name": "ophidia",
+                "image": self.ophidia_image,
+                "volumeMounts": ophidia_mounts,
+            }
+            spawner.extra_containers.append(ophidia)
```
Comment (andrea-manzi) [inline egi_notebooks_hub/d4science.py:499]: shouldn't this go after setting the  environment ? ( not sure)

Comment (enolfc) [reply egi_notebooks_hub/d4science.py:499]: you mean the lines below? should be the same as the container will be created in another function, here we are just preparing things

Comment (andrea-manzi) [reply egi_notebooks_hub/d4science.py:499]: ok !

**Review by andrea-manzi — APPROVED** [review]
LGTM!

**Pull request merged**